baseapp/baseapp.go

```diff
@@ -1007,6 +1007,25 @@ func (app *BaseApp) runTx(ctx sdk.Context, mode runTxMode, tx sdk.Tx, checksum [
 	if ctx.CheckTxCallback() != nil {
 		ctx.CheckTxCallback()(ctx, err)
 	}
+	var evmTxInfo *abci.EvmTxInfo
+	if ctx.IsEVM() {
+		evmTxInfo = &abci.EvmTxInfo{
+			SenderAddress: ctx.EVMSenderAddress(),
+			Nonce:         ctx.EVMNonce(),
+			TxHash:        ctx.EVMTxHash(),
+			VmError:       result.EvmError,
+		}
+	}
+	var events []abci.Event = []abci.Event{}
+	if result != nil {
```

Comment:
don't we also need to handle the case in which `result.EvmError` is non-nil? or do we want to index events for that case as well?

Reply:
the existing behavior is to index events even if EvmError is non-nil https://github.com/sei-protocol/sei-cosmos/pull/543/files#diff-389d02f089ab55575400f5300d3c5cfe277fde5f04bd0fe6a85456a7252d1099R309-R330

not sure if we want to change that. I can see reasons both ways